tests/components/sonos/test_media_player.py

```diff
@@ -13,10 +13,14 @@
 async def test_async_setup_entry_hosts(hass, config_entry, config, soco):
     """Test static setup."""
     await setup_platform(hass, config_entry, config)
-    assert hass.data[media_player.DATA_SONOS].entities[0].soco == soco
+
+    entity = hass.data[media_player.DATA_SONOS].entities[0]
+    assert entity.soco == soco
 
 
 async def test_async_setup_entry_discover(hass, config_entry, discover):
     """Test discovery setup."""
     await setup_platform(hass, config_entry, {})
-    assert hass.data[media_player.DATA_SONOS].uids == {'RINCON_test'}
+
+    entity = hass.data[media_player.DATA_SONOS].entities[0]
+    assert entity.unique_id == 'RINCON_test'
```

Comment:
We should avoid asserting things about entitites directly. Primarily use the state of the entity to assert when the state holds what we need to test.  For this case, we can check the unique_id by getting the entity registry entry for the entity, and assert that.

The idea is by avoiding using the entity directly in our tests, the code under test can be refactored without needing to change tests. This is very useful and will decrease risk of bugs.

Reply:
Thank you, I will keep that in mind.